fahrschein-http-test-support/src/main/java/org/zalando/fahrschein/http/test/AbstractRequestFactoryTest.java

```diff
@@ -0,0 +1,170 @@
+package org.zalando.fahrschein.http.test;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mockito;
+import org.zalando.fahrschein.http.api.ContentEncoding;
+import org.zalando.fahrschein.http.api.ContentType;
+import org.zalando.fahrschein.http.api.Request;
+import org.zalando.fahrschein.http.api.RequestFactory;
+import org.zalando.fahrschein.http.api.Response;
+
+import java.io.BufferedReader;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.stream.Collectors;
+import java.util.zip.GZIPInputStream;
+import java.util.zip.GZIPOutputStream;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.CoreMatchers.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertEquals;
+
+public abstract class AbstractRequestFactoryTest {
+
+    public abstract RequestFactory defaultRequestFactory(ContentEncoding contentEncoding);
+
+    protected static HttpServer server;
+    protected static URI serverAddress;
+
+    @BeforeClass
+    public static void startServer() throws IOException {
+        server = HttpServer.create(new InetSocketAddress("localhost", 0), 1);
+        serverAddress = URI.create("http://localhost:" + server.getAddress().getPort());
+        ExecutorService executor = Executors.newSingleThreadExecutor();
+        server.setExecutor(executor);
+        server.start();
+    }
+
+    @Captor
+    public ArgumentCaptor<HttpExchange> exchangeCaptor;
+
+    @Test
+    public void testGzippedResponseBody() throws IOException {
```

Comment:
It seems to me that we are not testing that the http implementations can deal with `IdentityAcceptEncodingRequestFactory`.